frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -1184,7 +1126,41 @@ export default ({
         e?.dataTransfer.files?.length &&
           this.$refs.sendArea.fileAttachmentHandler(e?.dataTransfer.files, true)
       }
-    }
+    },
+    async _processSwitchQueue () {
+      if (this.ephemeral.chatroomSwitchQueue.length === 0) return
+
+      // Take the most recent chatroom entry on the queue and discard everything else. This way we can speed up the process of switching chatrooms.
+      const targetChatroomId = this.ephemeral.chatroomSwitchQueue.pop()
+      this.ephemeral.chatroomSwitchQueue = []
+      this.ephemeral.renderingChatRoomId = targetChatroomId
+
+      try {
+        await this.initializeState(true)
+        if (this.ephemeral.chatroomSwitchQueue.length > 0) {
+          // If the user has since switched to another chatroom while initializing this chatroom, stop here
+          // and care about the switched chatroom.
+          return this.processSwitchQueue()
+        } else {
+          this.ephemeral.messagesInitiated = false
+          this.ephemeral.unprocessedEvents = []
+          this.ephemeral.infiniteLoading?.reset()
+        }
+      } catch (e) {
+        console.error('ChatMain.vue processSwitchQueue() error:', e)
+
+        if (!this.chatroomHasSwitchedFrom(targetChatroomId)) {
+          sbp('gi.ui/prompt', {
+            heading: L('Failed to initialize chatroom'),
+            question: L('Error details: {reportError}', LError(e)),
+            primaryButton: L('Close')
+          })
+        }
+      }
+    },
+    processSwitchQueue: debounce(function () {
+      this._processSwitchQueue()
+    }, 250)
```

Comment:
There's no reason to create a separate `_processSwitchQueue` function, you can put its contents inside here.

Reply:
Right. updated.